src/hooks/video/agora/useAgoraScreenShare.ts

```diff
@@ -68,24 +66,33 @@ export const useAgoraScreenShare: ReactHook<
       screenClientUid: `${screenClientUid}`,
     };
 
-    updateTalkShowStudioExperience({ venueId, userId, experience });
+    await updateTalkShowStudioExperience({ venueId, userId, experience });
   };
 
   const leaveChannel = useCallback(async () => {
-    stopShare();
+    await stopShare();
     await client?.leave();
   }, [client, stopShare]);
 
   useEffect(() => {
+    setClient(
+      AgoraRTC.createClient({
+        codec: "h264",
+        mode: "rtc",
+      })
+    );
```

Comment:
same

Reply:
@mike-lvov 
As a variant, after the calling to `createClient` function checking if we have client or not, if not so use Bugsnag. I didn't find any other extra API, only this one but if I understand correctly we are talking about the connection of the client to the channel and all info is sending to the servers of Agora. https://docs.agora.io/en/All/API%20Reference/web/modules/agorartc.logger.html#enablelogupload